frame/support/procedural/src/construct_runtime/expand/config.rs

```diff
@@ -0,0 +1,123 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+use crate::construct_runtime::Pallet;
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use syn::Ident;
+
+pub fn expand_outer_config(
+	runtime: &Ident,
+	pallet_decls: &[Pallet],
+	scrate: &TokenStream,
+) -> TokenStream {
+	let mut types = TokenStream::new();
+	let mut fields = TokenStream::new();
+	let mut build_storage_calls = TokenStream::new();
+
+	for decl in pallet_decls {
+		if let Some(pallet_entry) = decl.find_part("Config") {
+			let config = format_ident!("{}Config", decl.name);
+			let mod_name = decl.pallet.mod_name();
+			let field_name = if let Some(inst) = decl.instance.as_ref() {
+				format_ident!("{}_{}", mod_name, inst)
+			} else {
+				mod_name
+			};
+			let part_is_generic = !pallet_entry.generics.params.is_empty();
+
+			types.extend(expand_config_types(runtime, decl, &config, part_is_generic));
+			fields.extend(quote!(pub #field_name: #config,));
+			build_storage_calls.extend(expand_config_build_storage_call(scrate, runtime, decl, &field_name));
+		}
+	}
+
+	quote!{
+		#types
+
+		#[cfg(any(feature = "std", test))]
+		use #scrate::serde as __genesis_config_serde_import__;
+		#[cfg(any(feature = "std", test))]
+		#[derive(#scrate::serde::Serialize, #scrate::serde::Deserialize, Default)]
+		#[serde(rename_all = "camelCase")]
+		#[serde(deny_unknown_fields)]
+		#[serde(crate = "__genesis_config_serde_import__")]
```

Comment:
isn't it possible to use `#scrate::serde` directly here ?
But we can do as a polishing follow-up

Reply:
Oh yeah, I was just copy and pasting the code from the old macros and didn't really think about whether we needed to prepend `#scrate`. My thinking was that if it worked in the old macro verbatim, there's no reason why it wouldn't work on the new proc macros.

Reply:
It's interesting how using `#scrate::serde` actually produces 2 errors:

```
error[E0433]: failed to resolve: could not find `serde` in `hidden_include`
   --> frame/support/test/tests/construct_runtime.rs:223:1
    |
223 | / frame_support::construct_runtime!(
224 | |     pub enum Runtime where
225 | |         Block = Block,
226 | |         NodeBlock = Block,
...   |
242 | |     }
243 | | );
    | |__^ could not find `serde` in `hidden_include`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `serde` in `hidden_include`
   --> frame/support/test/tests/construct_runtime.rs:278:2
    |
278 | /     frame_support::construct_runtime!(
279 | |         pub enum RuntimeOriginTest where
280 | |             Block = Block,
281 | |             NodeBlock = Block,
...   |
287 | |         }
288 | |     );
    | |______^ could not find `serde` in `hidden_include`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Because you are deriving 2 traits. Nothing special here